File: Leetcode/hashmap/739_Daily Temperatures/stack_solution/Solution.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& T) {
        stack<int> stk;
        vector<int> res(T.size());
        for(int i = 0; i<T.size(); ++i){
            while(!stk.empty() && T[stk.top()]<T[i]){
                int idx = stk.top();
                stk.pop();
                res[idx] = i - idx;
            }
            stk.push(i);
        }
        return res;
    }
};

File: Leetcode/hashmap/739_Daily Temperatures/stack_solution/Solution.py
class Solution(object):
    def dailyTemperatures(self, T):
        """
        :type T: List[int]
        :rtype: List[int]
        """
        stack = []
        res = [0]*len(T)
        for i in range(len(T)):
            while stack and T[stack[-1]]<T[i]:
                idx = stack.pop()
                res[idx] = i - idx
            stack.append(i)
        return res
